backend/src/shared/validators/is-before.validator.ts

```diff
@@ -0,0 +1,17 @@
+import {
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+  ValidationArguments,
+} from 'class-validator';
+
+// https://github.com/typestack/class-validator/issues/145
+@ValidatorConstraint({ name: 'isBefore', async: false })
+export class IsBeforeConstraint implements ValidatorConstraintInterface {
```

Comment:
found a neat way to validate that a date is before another date using class-validator :)